internal/testhelp/utils.go

```diff
@@ -27,12 +28,32 @@ func RandomName(length ...int) string {
 	return acctest.RandStringFromCharSet(size, "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
 }
 
+// RandomInt returns a random integer between minInt (inclusive) and maxInt (exclusive).
+func RandomInt(minInt, maxInt int32) int32 {
+	// #nosec G115 -- safe cast because we know minInt/maxInt fit in int32
+	return int32(acctest.RandIntRange(int(minInt), int(maxInt)))
+}
```

Comment:
* RandomIntRange better shows intention of the methot
* refactor idea: make it generic to have option to use any type of int

```suggestion
// RandomIntRange returns a random integer between minInt (inclusive) and maxInt (exclusive).
func RandomIntRange[T ~int | ~int32 | ~int64](minInt, maxInt T) T {
	if minInt >= maxInt {
		panic(fmt.Sprintf("minInt %d must be less than maxInt %d", minInt, maxInt))
	}

	// Ensure that minInt and maxInt are of the same type
	if minIntType := fmt.Sprintf("%T", minInt); fmt.Sprintf("%T", maxInt) != minIntType {
		panic(fmt.Sprintf("minInt and maxInt must be of the same type, got %T and %T", minInt, maxInt))
	}
	// Generate a random integer in the range [minInt, maxInt]
	switch minInt := any(minInt).(type) {
	case int:
		return T(rand.Intn(int(maxInt)-minInt) + minInt)
	case int32:
		return T(rand.Int31n(int32(maxInt)-minInt) + minInt)
	case int64:
		return T(rand.Int63n(int64(maxInt)-minInt) + minInt)
	default:
		panic(fmt.Sprintf("unsupported type %T", minInt))
	}
}
```